### PR TITLE
handle buffer offset

### PR DESCRIPTION
**Pull request by cmeissl**
tracking of https://github.com/Smithay/smithay/pull/1341

### CODE REVIEW

**Comment by cmeissl** [comment]
@YaLTeR This is now just waiting for the review of https://github.com/Smithay/smithay/pull/1341

File: src/handlers/compositor.rs [hunk 1]
```diff
@@ -297,13 +297,42 @@ impl CompositorHandler for State {
             &self.niri.cursor_manager.cursor_image(),
             CursorImageStatus::Surface(s) if s == &root_surface
         ) {
+            with_states(surface, |states| {
```
Comment (YaLTeR) [inline src/handlers/compositor.rs:300]: Guess this should only happen if `surface == root_surface`

File: src/handlers/compositor.rs [hunk 2]
```diff
@@ -297,13 +297,42 @@ impl CompositorHandler for State {
...
             // FIXME: granular redraws for cursors.
             self.niri.queue_redraw_all();
             return;
         }
 
         // This might be a DnD icon surface.
-        if self.niri.dnd_icon.as_ref() == Some(&root_surface) {
+        if matches!(&self.niri.dnd_icon, Some(icon) if icon.surface == root_surface) {
+            let dnd_icon = self.niri.dnd_icon.as_mut().unwrap();
+            with_states(&dnd_icon.surface, |states| {
```
Comment (YaLTeR) [inline src/handlers/compositor.rs:326]: Guess this can only change if surface == root_surface

File: src/handlers/compositor.rs (hunk 1 above)
Comment (cmeissl) [reply src/handlers/compositor.rs]: right, while unlikely the surface could be part of a surface tree

File: src/input/mod.rs [hunk 3]
```diff
@@ -2754,8 +2750,8 @@ mod tests {
         // The key_code we pick is arbitrary, the only thing
         // that matters is that they are different between cases.
 
-        let close_key_code = close_keysym.into();
-        let close_key_event = |suppr: &mut HashSet<u32>, mods: ModifiersState, pressed| {
+        let close_key_code =  Keycode::from(close_keysym.raw() + 8u32);
```
Comment (cmeissl) [inline src/input/mod.rs:2753]: Not sure about the offset here tbh

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**